source/server/lds_api.cc

```diff
@@ -50,13 +50,23 @@ void LdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
   // We need to keep track of which listeners we might need to remove.
   std::unordered_map<std::string, std::reference_wrapper<Network::ListenerConfig>>
       listeners_to_remove;
+
   for (const auto& listener : listener_manager_.listeners()) {
     listeners_to_remove.emplace(listener.get().name(), listener);
   }
 
+  for (const auto& listener : resources) {
+    listeners_to_remove.erase(listener.name());
+  }
+
+  for (const auto& listener : listeners_to_remove) {
+    if (listener_manager_.removeListener(listener.first)) {
+      ENVOY_LOG(info, "lds: remove listener '{}'", listener.first);
+    }
+  }
```

Comment:
Can you add some comments explaining this ordering and warning against changing?

Reply:
Done.